src/simulator/player.py

```diff
@@ -0,0 +1,205 @@
+    STAY = 0
+    HIT = 1
+    SPLIT = 2
+    DOUBLE = 3
+
+
+class BasePlayerHand:
+    """
+    Base player hand class to inherit from to enact policy decisions
+    NOTE: dealer will also inherit from this
+    """
+
+    def __init__(self, deck):
+        self.deck = deck
+
+        hand = Hand(deck.next_card(), deck.next_card())
+        self.hands = [hand]
+        self.curr_hand_ndx = 0
+
+    def play_hand(self):
+        """
+        Performs actions for the hand based on the policy decided by
+        self.get_action until the player busts or stays for each hand
+        :return:
+        """
+        while self.curr_hand_ndx < len(self.hands):
+            self.perform_action(
+                self.get_action(
+                    self.hands[self.curr_hand_ndx]
+                )
+            )
+
+    def get_action(self, hand):
+        """
+        Driven by the policy of the Player
+        :param hand: Hand that the player is on
+                     NOTE: This can be found easily with instance variables,
+                           but passing it as an argument will save time
+        :return: PlayerAction enum value
+        """
+        raise NotImplementedError()
+
+    def perform_action(self, action):
+        """
+        Performs the action specified by the enum
+
+        NOTE: If needed for args, we can remove this middle man by having the
+              self.get_action function just call self.hit etc. directly
+        :param action: PlayerAction enum
+        :return:
+        """
+        if action == PlayerAction.STAY:
+            self._stay()
+        elif action == PlayerAction.HIT:
+            self._hit()
+        elif action == PlayerAction.SPLIT:
+            self._split()
+        elif action == PlayerAction.DOUBLE:
+            self._double()
+
+    def _stay(self):
+        self.curr_hand_ndx += 1
+
+    def _hit(self):
+        try:
+            self.hands[self.curr_hand_ndx].add_card(
+                self.deck.next_card()
+            )
+        except BustException:
+            self.curr_hand_ndx += 1
+
+    def _double(self):
+
+        if len(self.hands[self.curr_hand_ndx].cards) > 2:
+            raise ValueError("Can't double after the initial deal")
+
+        try:
+            self.hands[self.curr_hand_ndx].add_card(
+                self.deck.next_card(),
+                double=True
+            )
+        except BustException:
+            # Doubled, so we're done with the hand regardless
+            pass
+
+        self.curr_hand_ndx += 1
+
+    def _split(self):
+        old_hand = self.hands[self.curr_hand_ndx]
+        # Figured we may as well throw this validation here
+        if len(old_hand.cards) != 2 or old_hand.cards[0] != old_hand.cards[1]:
+            raise ValueError("Can't split on non-matching hand")
+
+        self.hands[self.curr_hand_ndx] = Hand(
+            old_hand.cards[0],
+            self.deck.next_card()
+        )
+        self.hands.insert(
```

Comment:
I think you might just be able to use the "append()" function here

Reply:
I thought that too, but we can split on the first hand of a split unfortunately